src/attributes.jl

```diff
@@ -317,6 +331,13 @@ struct ResultCount <: AbstractInstanceAttribute end
 
 ## Variable attributes
 
+"""
+    ListOfVariableAttributesSet()
+
+A `Vector{AbstractVariableAttribute}` of all variable attributes that were set to the variable.
+"""
+struct ListOfVariableAttributesSet <: AbstractVariableAttribute end
```

Comment:
This should be an instance attribute instead of a variable attribute. It could be expensive to query this for every variable. The definition of `canget` says:
    
    Return a Bool indicating whether the instance instance currently has a value for the attributed specified by attribute type attr applied to every variable indexed by v or constraint indexed by c.

Any attribute that's available for one variable has to be available for all. This also applies to constraints, but we may need to revise this definition to restrict it to all constraints of the same `F`-in-`S` type.

Reply:
Indeed, good point.
Once we have the list, we still need to do `MOI.get` for every attribute and every attribute but I guess this is unavoidable.

Reply:
You mean every attribute and every variable? We have the vector version of `get`.

Reply:
Ah yes right. 
There is also the `canget` version for single variable. If an attribute is only set to a few variables, calling `MOI.canget(instance, attr, MOI.get(instance, MOI.ListOfVariableIndices())`, will return false because it is not applied to all variables so they will be ignored by `MOI.copy!`.

Reply:
I don't think we should allow attributes to be defined only for some variables.

Reply:
I have updated the PR so that the list of attributes does not depend on the variable. As far as this PR is concerned this solve the issue. It remains to decide what we do once we have the list of attributes but this is not related to this PR so I have moved the discussion to https://github.com/JuliaOpt/MathOptInterface.jl/issues/191